scalyr_agent/util.py

```diff
@@ -1257,6 +1257,25 @@ def charge_if_available(self, num_bytes, current_time=None):
 
         @return: True if there are enough room in the rate limit to allow the operation.
         """
+        if self._get_time_to_sleep(num_bytes, current_time) > 0.0:
+            return False
+        else:
+            self.__bucket_contents -= num_bytes
+            return True
+
+    def sleep_and_charge(self, num_bytes, current_time=None):
```

Comment:
I would call this `block_until_charge_succeeds`.  It's a bit wordy, but probably better to use `block` instead of `sleep` (it's a bit of an implementation detail we sleep) and I wanted to make it a little clearer as to what we are blocking on.